### PR TITLE
Enable replicasets and deployments

### PR DESCRIPTION
**Pull request by deads2k**
Enables RSs and Ds without cohabitation.

@mfojtik

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 1]
```diff
@@ -316,7 +316,7 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 
 				authorizationapi.NewRule(read...).Groups(batchGroup).Resources("jobs").RuleOrDie(),
 
-				authorizationapi.NewRule(read...).Groups(extensionsGroup).Resources("jobs", "horizontalpodautoscalers").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(extensionsGroup).Resources("jobs", "horizontalpodautoscalers", "replicasets", "replicasets/scale").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: add to cluster reader

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: test to catch drift there would be good

Comment (deads2k) [inline pkg/cmd/server/bootstrappolicy/policy.go]: > test to catch drift there would be good

It exists.  It'll yell in a minute.

Comment (deads2k) [inline pkg/cmd/server/bootstrappolicy/policy.go]: > add to cluster reader

Also, it already exists in cluster-reader.

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: ah, missed that

**Comment by stevekuznetsov** [comment]
re[test]

File: pkg/cmd/server/bootstrappolicy/infra_sa_policy.go [hunk 2]
```diff
@@ -183,6 +189,45 @@ func init() {
...
 
 	err = InfraSAs.addServiceAccount(
+		InfraDeploymentControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: DeploymentControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				{
+					APIGroups: []string{"extensions"},
```
Comment (mfojtik) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: `extensions.GroupName`

**Comment by mfojtik** [comment]
LGTM (I think we should have some test-cmd as well to prove the scale works, but I can do that as follow up).

**Comment by deads2k** [comment]
[merge]

File: pkg/cmd/server/bootstrappolicy/infra_sa_policy.go [hunk 3]
```diff
@@ -149,10 +155,10 @@ func init() {
 	}
 
 	err = InfraSAs.addServiceAccount(
-		InfraDeploymentControllerServiceAccountName,
+		InfraDeploymentConfigControllerServiceAccountName,
 		authorizationapi.ClusterRole{
 			ObjectMeta: kapi.ObjectMeta{
-				Name: DeploymentControllerRoleName,
+				Name: DeploymentConfigControllerRoleName,
```
Comment (kargakis) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: This is not the deployment config controller but our custom replication controller controller

**Pull request merged**